### PR TITLE
Add README for cryptol-remote-api

### PR DESCRIPTION
**Pull request by david-christiansen**
This consists of a mix of README content migrated from the argo repo and fresh content that was missing here.

I'd appreciate one of the reviewers merging this PR if it gets approved, as I'm out for the next couple of weeks.

### CODE REVIEW

**Review by ChrisEPhifer — COMMENTED** [review]
This looks good to me, though I think someone else should confirm the Python/Emacs sections are correct (given my limited familiarity with the details of those components).

File: cryptol-remote-api/README.rst [hunk 1]
```diff
@@ -0,0 +1,190 @@
+``cryptol-remote-api``
+======================
+
+This is the JSON-RPC server for Cryptol. It is primarily intended for
+using an executable Cryptol specification as part of a prototype or a
+system in which performance is less important. It supports loading and
+type checking Cryptol modules,
```
Comment (atomb) [inline cryptol-remote-api/README.rst:7]: Is this sentence unfinished, or should that be a period?

File: cryptol-remote-api/README.rst [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+--------------------------
+
+The protocol description and documentation are available in the
+``docs`` subdirectory. The documentation is in Sphinx format. Run
+``make html`` to generate readable HTML.
+
+Obtaining the Python bindings
+-----------------------------
+
+The Python bindings for ``crytpol-remote-api`` are presently part of
```
Comment (atomb) [inline cryptol-remote-api/README.rst:31]: `crytpol` -> `cryptol`

**Review by atomb — APPROVED** [review]
I noticed a couple of small typos, and I haven't tested the Emacs interaction, but otherwise this looks good. Thanks!

File: cryptol-remote-api/README.rst [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+2. Open ``proto-test.el``
+
+3. Evaluate the buffer: ``M-x eval-buffer`` or on Spacemacs: ``, e b``
+
+To use the stdio version:
+
+1. ``M-x proto-test-start``
+
+2. At the prompt for ``Command:``, run the server with ``cabal v2-exec -v0
+   saw-remote-api`` or ``cabal v2-exec -v0 cryptol-remote-api``.
```
Comment (pnwamk) [inline cryptol-remote-api/README.rst:155]: I'm not sure if this is relevant on this line, but don't both server APIs need one of `http`, `socket`, or `stdio` following their name to run them?

File: cryptol-remote-api/README.rst [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+If this leaves a confusing error message in Emacs, the output was
+probably corrupted by ``cabal-install`` stating that nothing needs
+building. Run ``cabal v2-build all`` to make sure that all builds are
+up-to-date, and try again.
+
+
+To use the socket version:
+
+1. At a shell, run ``cabal v2-exec cryptol-remote-api -- --port 10006``
```
Comment (pnwamk) [inline cryptol-remote-api/README.rst:165]: `cabal v2-exec cryptol-remote-api -- --port 10006` => `cabal v2-exec cryptol-remote-api -- socket --port 10006` (i.e., missing `socket`)

**Review by pnwamk — APPROVED** [review]
Looks good to me modulo the two comments I made.

**Comment by david-christiansen** [comment]
Would you mind making the changes and merging? I'm off for two weeks and it
seems that the incorrect README is actively causing issues for people.

Thanks!

**Pull request merged**